api/envoy/data/tap/v2alpha/wrapper.proto

```diff
@@ -16,6 +18,17 @@ option java_package = "io.envoyproxy.envoy.data.tap.v2alpha";
 // Wrapper for all fully buffered and streamed tap traces that Envoy emits. This is required for
 // sending traces over gRPC APIs or more easily persisting binary messages to files.
 message TraceWrapper {
+  message Destination {
+    string cluster_name = 1;
+    google.protobuf.Struct cluster_metadata = 2;
+    envoy.api.v2.core.Address host_address = 3;
+    google.protobuf.Struct host_metadata = 4;
+  }
+  // if trace is streaming, this may only be sent once.
+  Destination destination = 5;
+
+  envoy.api.v2.core.Address downstream_remote_address = 6;
```

Comment:
This may not always apply if the tap is for an upstream connection. Can we future proof with this with maybe a message covering downstream connection info, upstream connection info, etc.?